### DISCUSSION TITLE
HmIP-ESI-GAS gas counter

### DISCUSSION THREAD

**Discussion by Isengo1989** [SukramJ/aiohomematic#1445] [Q&A] [answered]
Short question, how long does it typically take till a new device is supported? Currently eyeballing the HmIP-ESI-GAS for pre-order.

**Comment by SukramJ** [comment, marked as answer]
Every device is supported [immediatly](https://github.com/danielperna84/custom_homematic#device-support).
If not, then some one needs to [support](https://github.com/danielperna84/custom_homematic#homematicip_localexport_device_definition) the project.

**Reply by Boarder** [reply]
I could support. I have the device and unfortunately it does not show up in HA.
What ever you need to know I would like to assist.

**Reply by SukramJ** [reply]
Search for the device type in the device list.
The device is created in HA before being renamed to CCU.

**Reply by Boarder** [reply]
oh man. the device was already listed under my HM devices with its correct name and not with its serial number. Sorry for that

**Reply by Boarder** [reply]
<img width="527" alt="Gaszähler" src="https://github.com/danielperna84/hahomematic/assets/45979973/52f83385-bc51-4ab3-828c-3d2353631676">
but unfortunately I cannot add one of these entity's to my energy dashboard.....maybe you know how? Thanks

**Reply by SukramJ** [reply]
You need to add a [state_class](https://developers.home-assistant.io/docs/core/entity/sensor#available-state-classes) via [customize](https://www.home-assistant.io/docs/configuration/customizing-devices/#manual-customization).
What is the different between Gas Flow and Gas Volume? Are these values just increasing?

**Reply by Boarder** [reply]
Gas Flow is going up and down and Gas Volume is just increasing

**Reply by SukramJ** [reply]
..., then state_class for flow is `measurement` and  for volume is `total`.
I will add that to the next release, but that could take a while. In the meantime [customize](https://www.home-assistant.io/docs/configuration/customizing-devices/#manual-customization) is your friend.

**Reply by SukramJ** [reply]
Please attach the device definition as described [here](https://github.com/danielperna84/custom_homematic#homematicip_localexport_device_definition).

**Reply by Boarder** [reply]
With the new version it is work now. THX 👍

**Reply by SukramJ** [reply]
> Please attach the device definition as described [here](https://github.com/danielperna84/custom_homematic#homematicip_localexport_device_definition).

The device descriptions would still be very helpful!

**Comment by Boarder** [comment]
HiGerne würde ich hier unterstützen und die Device Infos hinterlegen, aber obwohl ich mir die Seite s. U. 2 mal durchgelesen habe, erschließt es sich mir nicht, was ich wie wo hinterlegen könnteSorryGrußGregorAm 04.03.2024 um 11:30 schrieb SukramJ ***@***.***>:﻿

Please attach the device definition as described here.

The device descriptions would still be very helpful!

**Reply by SukramJ** [reply]
Also du führst in HA den Service  (Entwickler Werkzeuge --> Dienste) aus, und gehst dann ins Dateisystem, nimmst die beiden Dateien, am besten gezippt, und hängst die hier an, oder noch besser als Pull request by pydevccu.

**Comment by Boarder** [comment]
welchen den? Wenn ich unter Dienst Service eingebe kommt nix....mit home* diese Liste:

**Reply by SukramJ** [reply]
Einfach `homematicip_local.export_device_definition` copy&paste. Ein * funktioniert nicht

**Comment by Boarder** [comment]
so geschafft, anbei die beiden gewünschten Dateien.
da ich mich erst in die Erstellung von Pull requests einarbeiten muss, schicke ich sie lieber schon mal vorab
 
Vielen Dank für Deine Unterstützung und Mega Arbeit mit Deinem Plugin ohne wäre HA für mich sinnlos
 
Kann man Dich mit eine mit einem Kaffee (oder Bier) unterstützen ?

**Reply by SukramJ** [reply]
> Kann man Dich mit eine mit einem Kaffee (oder Bier) unterstützen ?

Danke, nicht nötig.

Du kannst die Dateien einfach hier anhängen.

**Reply by SukramJ** [reply]
Solltest Du versucht habe die Dateien als Mail Anhang hochzuladen, so funktioniert das leider nicht.

**Reply by Boarder** [reply]
[HmIP-ESI.zip](https://github.com/danielperna84/hahomematic/files/14505669/HmIP-ESI.zip)

ich hoffe, dass die Datei jetzt dabei ist...

**Reply by SukramJ** [reply]
Sehr gut, vielen Dank
